core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +136,30 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
-            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
-                    stack.getName(), stack.getStatus()));
+    public void validateStackStatusForStartHostGroup(StackDto stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentV4Request) {
+
+        if (stack.getStack().isModificationInProgress()) {
+            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Upscaling is not allowed.",
+                    stack.getStack().getName(), stack.getStack().getStatus()));
+        }
+
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
+        Set<String> dependentComponents = processor.getNonGatewayComponentsByHostGroup().get(instanceGroupAdjustmentV4Request.getInstanceGroup()).stream()
+                .map(srcComponent -> roleDependencyService.getRoleDependency(srcComponent))
+                .flatMap(Collection::stream)
+                .collect(toSet());
+        List<String> unhealthyHostGroupNames = processor.getUnhealthyDependentHostGroups(stack, dependentComponents,
+                UNDEFINED_DEPENDENCY);
+
+        if (!unhealthyHostGroupNames.isEmpty()) {
+            if (unhealthyHostGroupNames.size() > 1 || !unhealthyHostGroupNames.contains(UNDEFINED_DEPENDENCY)) {
```

Comment:
Don't understand the > 1 check

Reply:
If there is master (for the case of nodemanager) which is unhealthy and there is some component as well whose dependency is not defined than in that case we will have master and UNDEFINED_DEPENDENCY in unhealthyHostGroupNames. 
So, now as we have undefined_dependency and we do the regular check and if the stack is avaiable than we would proceed with upscaling but we should not proceed as we still have nodemanager whose dependent host is unhealthy.
So, I have added this check.